packages/web/app/views/patients/medication/MarTable.jsx

```diff
@@ -0,0 +1,271 @@
+import { useEncounterMedicationQuery } from '../../../api/queries/useEncounterMedicationQuery';
+import { format, parse } from 'date-fns';
+import { getDateFromTimeString } from '@tamanu/shared/utils/medication';
+
+const Container = styled.div`
+  position: relative;
+`;
+
+const MedicationContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  max-height: calc(100vh - 221px);
+  background-color: ${Colors.white};
+`;
+
+// Header row for the time slots
+const HeaderRow = styled.div`
+  display: grid;
+  grid-template-columns: minmax(100px, 1fr) repeat(${props => props.columns}, 48px);
+  position: sticky;
+  top: 0;
+  z-index: 10;
+  background-color: ${Colors.white};
+`;
+
+const ScrollableContent = styled.div`
+  overflow-y: auto;
+  ${p => (p.$flexShrink || p.$flexShrink === 0) && `flex-shrink: ${p.$flexShrink};`}
+`;
+
+const HeadingCell = styled.div`
+  display: flex;
+  align-items: center;
+  padding: 12px;
+  font-size: 16px;
+  font-weight: 500;
+  color: ${Colors.darkestText};
+  border-top: 1px solid ${Colors.outline};
+  border-left: 1px solid ${Colors.outline};
+  height: 100%;
+`;
+
+const TimeSlotHeaderContainer = styled.div`
+  padding: 24px 0px 10px 0px;
+  height: 105px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  border-top: 1px solid ${Colors.outline};
+  border-left: 1px solid ${Colors.outline};
+  ${props => props.isCurrentTimeSlot && `background: #EBF0F5; color: ${Colors.primary};`}
+`;
+
+const TimeSlotText = styled.div`
+  font-weight: 400;
+  font-size: 12px;
+  transform: rotate(-90deg);
+  white-space: nowrap;
+  color: ${Colors.midText};
+`;
+
+const TimeSlotLabel = styled.div`
+  text-transform: capitalize;
+  color: ${Colors.darkestText};
+`;
+
+const SubHeader = styled.div`
+  font-size: 14px;
+  font-weight: 500;
+  color: ${Colors.midText};
+  padding: 10px 8px;
+  border-top: 1px solid ${Colors.outline};
+  border-left: 1px solid ${Colors.outline};
+  border-bottom: 1px solid ${Colors.outline};
+  margin-top: -1px;
+  grid-column: 1 / -1;
+  position: sticky;
+  top: 0;
+  z-index: 5;
+  background-color: ${Colors.white};
+`;
+
+const EmptyMessage = styled.div`
+  color: ${Colors.darkestText};
+  font-size: 14px;
+  font-weight: 500;
+  padding: 10px 8px;
+  border-top: 1px solid ${Colors.outline};
+  border-left: 1px solid ${Colors.outline};
+  grid-column: 1 / -1;
+`;
+
+const MedicationCellContainer = styled.div`
+  padding: 8px 12px;
+  font-size: 14px;
+  border-top: 1px solid ${Colors.outline};
+  border-left: 1px solid ${Colors.outline};
+  ${props => props.discontinued && `text-decoration: line-through;`}
+`;
+
+const StatusCell = styled.div`
+  border-top: 1px solid ${Colors.outline};
+  border-left: 1px solid ${Colors.outline};
+`;
+
+const MedicationGrid = styled.div`
+  display: grid;
+  grid-template-columns: minmax(100px, 1fr) repeat(12, 48px);
+`;
+
+const CurrentTimeOverlay = styled.div`
+  position: absolute;
+  top: 0;
+  width: 48px;
```

Comment:
We should put the 48px in a constant at the top of the file since it is important for the layout